nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/main/java/org/apache/nifi/processors/iceberg/converter/IcebergRecordConverter.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    private static class IcebergSchemaVisitor extends SchemaWithPartnerVisitor<DataType, DataConverter<?, ?>> {
+
+        public static DataConverter<?, ?> visit(Schema schema, RecordDataType recordDataType, FileFormat fileFormat) {
+            return visit(schema, recordDataType, new IcebergSchemaVisitor(), new IcebergPartnerAccessors(fileFormat));
+        }
+
+        @Override
+        public DataConverter<?, ?> schema(Schema schema, DataType dataType, DataConverter<?, ?> converter) {
+            return converter;
+        }
+
+        @Override
+        public DataConverter<?, ?> field(Types.NestedField field, DataType dataType, DataConverter<?, ?> converter) {
+            return converter;
+        }
+
+        @Override
+        public DataConverter<?, ?> primitive(Type.PrimitiveType type, DataType dataType) {
+            if (type.typeId() != null) {
+                switch (type.typeId()) {
+                    case BOOLEAN:
+                    case INTEGER:
+                    case LONG:
+                    case FLOAT:
+                    case DOUBLE:
+                    case DATE:
+                    case STRING:
+                        return GenericDataConverters.SameTypeConverter.INSTANCE;
+                    case TIME:
+                        return GenericDataConverters.TimeConverter.INSTANCE;
+                    case TIMESTAMP:
+                        Types.TimestampType timestampType = (Types.TimestampType) type;
+                        if (timestampType.shouldAdjustToUTC()) {
+                            return GenericDataConverters.TimestampWithTimezoneConverter.INSTANCE;
+                        }
+                        return GenericDataConverters.TimestampConverter.INSTANCE;
+                    case UUID:
+                        UUIDDataType uuidType = (UUIDDataType) dataType;
+                        if (uuidType.getFileFormat() == FileFormat.PARQUET) {
+                            return GenericDataConverters.UUIDtoByteArrayConverter.INSTANCE;
+                        }
+                        return GenericDataConverters.SameTypeConverter.INSTANCE;
+                    case FIXED:
+                        Types.FixedType fixedType = (Types.FixedType) type;
+                        return new GenericDataConverters.FixedConverter(fixedType.length());
+                    case BINARY:
+                        return GenericDataConverters.BinaryConverter.INSTANCE;
+                    case DECIMAL:
+                        Types.DecimalType decimalType = (Types.DecimalType) type;
+                        return new GenericDataConverters.BigDecimalConverter(decimalType.precision(), decimalType.scale());
+                    default:
+                        throw new UnsupportedOperationException("Unsupported type: " + type.typeId());
+                }
+            }
+            return null;
```

Comment:
The original method that is overridden returns with null by default, so that's why I followed this approach when `PrimitiveType` doesn't have a type id.

Reply:
I'm still uncertain about this part. I think we should handle the same way a non existent and an unknown type.